onnx/backend/test/case/node/nllloss.py

```diff
@@ -0,0 +1,191 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import numpy as np  # type: ignore
+
+import onnx
+from ..base import Base
+from . import expect
+
+
+def compute_nll_loss(input, target, weight=None, reduction='mean'):  # type: ignore
```

Comment:
Does it produce the same outputs as Pytorch's NLLLoss?

Reply:
# this is the test script to verify equivalency between pytorch and nllloss.py test data. 
def compute_nll_loss(input, target, weight=None, reduction='mean'):  # type: ignore
    ''' Compute nll_loss '''
    input_shape = input.shape

    # GatherElement(-input, target)
    if len(input_shape) == 2:
        N, C = input_shape
        neg_gather_element_input = np.zeros((N, ), dtype=np.float32)
        for i in range(N):
            neg_gather_element_input[i] = -input[i][target[i]]
    else:
        N, C, dim1, dim2 = input_shape
        neg_gather_element_input = np.zeros((N, dim1, dim2), dtype=np.float32)
        for i in range(N):
            for d1 in range(dim1):
                for d2 in range(dim2):
                    neg_gather_element_input[i][d1][d2] = -input[i][target[i][d1][d2]][d1][d2]

    loss = neg_gather_element_input
    if weight is not None:
        # Gather(input=weight, index=target)
        gather_weight = np.take(weight, target)

        loss = gather_weight * loss
        if reduction == 'mean':
            return loss.sum() / gather_weight.sum()

    if reduction == 'none':
        return loss
    elif reduction == 'mean':
        return np.mean(loss, keepdims=False)
    elif reduction == 'sum':
        return np.sum(loss, keepdims=False)

import numpy as np
from itertools import product
import torch
import torch.nn as nn
from numpy.testing import assert_allclose, assert_array_equal

# options 
reduction_options=['none', 'mean', 'sum']
N, C, height, width = 2, 3, 4, 5

sample_shape_option = [(), (height, width)]
weight_options=[None, np.random.randn(C).astype(np.float)]

for config in list(product(sample_shape_option, reduction_options, weight_options)):
    sample_shape, reduction, weight = config

    if weight is None:
        weight_pt = None
    else:
        weight_pt = torch.from_numpy(weight).float()

    loss = nn.NLLLoss(weight=weight_pt, reduction=reduction)
    log_softmax_ = nn.LogSoftmax(dim=1)

    data_ = torch.randn(N, C, *sample_shape)
    input_ = log_softmax_(data_)
    target = torch.empty((N, *sample_shape), dtype=torch.long).random_(0, C)

    input_.shape, target.shape

    output_pt = loss(input_, target)
    output_pt.shape, output_pt

    my_loss = compute_nll_loss(input_, target, weight=weight, reduction=reduction)
    assert_allclose(my_loss, output_pt.numpy(), rtol=1e-05)
    my_loss - output_pt.numpy()
    print ("pass: ", sample_shape, reduction, weight, my_loss - output_pt.numpy())